specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,208 @@
+            case COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION_BOTTOM_LEFT:
+                offset = {padding, -padding};
+                break;
+            case COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION_BOTTOM_RIGHT:
+                offset = {-padding, -padding};
+                break;
+        }
+        CHECK_FAILURE(m_webView2_6->SetDefaultDownloadDialogPosition(
+            targetPosition, offset));
+    }
+}
+```
+
+```c#
+void ToggleDownloadDialogCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    if (webView.CoreWebView2.IsDefaultDownloadDialogOpen)
+    {
+        webView.CoreWebView2.CloseDefaultDownloadDialog();
+    }
+    else
+    {
+        webView.CoreWebView2.OpenDefaultDownloadDialog();
+    }
+}
+
+void DownloadDialogPositionCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2DefaultDownloadDialogPosition newPosition =
+        CoreWebView2DefaultDownloadDialogPosition.TopRight;
+    System.Drawing.Point offset = new System.Drawing.Point();
+    string position = e.Parameter.ToString();
+    int padding = 20;
+    if (position == "Top-left")
+    {
+        newPosition =
+            CoreWebView2DefaultDownloadDialogPosition.TopLeft;
+        offset = new System.Drawing.Point(padding, padding);
+    }
+    else if (position == "Top-right")
+    {
+        newPosition =
+            CoreWebView2DefaultDownloadDialogPosition.TopRight;
+        offset = new System.Drawing.Point(-padding, padding);
+    }
+    else if (position == "Bottom-left")
+    {
+        newPosition =
+            CoreWebView2DefaultDownloadDialogPosition.BottomLeft;
+        offset = new System.Drawing.Point(padding, -padding);
+    }
+    else if (position == "Bottom-right")
+    {
+        newPosition =
+            CoreWebView2DefaultDownloadDialogPosition.BottomRight;
+        offset = new System.Drawing.Point(-padding, -padding);
+    }
+    webView.CoreWebView2.SetDefaultDownloadDialogPosition(
+        newPosition, offset);
+}
+```
+
+# API Details
+
+
+```c#
+/// The default download dialog can be aligned to any of the WebView corners.
+/// Use `SetDefaultDownloadDialogPosition` to specify a position and optional
+/// offset from that position.
+typedef enum COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION {
+
+  /// Top-left corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION_TOP_LEFT,
+
+  /// Top-right corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION_TOP_RIGHT,
+
+  /// Bottom-left corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION_BOTTOM_LEFT,
+
+  /// Bottom-right corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION_BOTTOM_RIGHT,
+
+} COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_POSITION;
+
+[uuid(9139c04d-8f37-42ae-8b63-01940c34d22f), object, pointer_default(unique)]
+interface ICoreWebView2_6 : ICoreWebView2_5
+{
+  /// `TRUE` if the default download dialog is currently open.
+  [propget] HRESULT IsDefaultDownloadDialogOpen(
```

Comment:
Edge's download button is fancy showing the progress of the download and such. Does it also change based on if the download dialog is open or not? I can see an IsDefaultDownloadDialogOpenChanged event potentially being useful.

Reply:
Ah yes the Edge button appears selected when the dialog is open. Sounds good, I can add an event

Reply:
To be clear, the position and padding properties will go on the event args?

Reply:
No, everything else as is, but then an additional event with IUnknown/nullptr event args that is raised whenever the value of IsDefaultDownloadDialogOpen changes. We won't need an event args since the caller can just check the CoreWebView2.IsDefaultDownloadDialogOpen property.

Reply:
OK got it, thanks!